nominal/cli/util/global_decorators.py

```diff
@@ -137,11 +146,25 @@ def wrapped_function(
         *args: Param.args,
         **kwargs: Param.kwargs,
     ) -> T:
+        profile: str | None = kwargs.pop("profile")  # type: ignore[assignment]
         base_url: str = kwargs.pop("base_url", "")  # type: ignore[assignment]
         token: str | None = kwargs.pop("token")  # type: ignore[assignment]
         token_path: pathlib.Path = kwargs.pop("token_path")  # type: ignore[assignment]
         trust_store_path: pathlib.Path | None = kwargs.pop("trust_store_path")  # type: ignore[assignment]
 
+        trust_store_str = str(trust_store_path) if trust_store_path else None
+
+        if profile is not None:
+            logger.info("Instantiating client from profile '%s'", profile)
+            client = NominalClient.from_profile(profile, trust_store_path=trust_store_str)
+            kwargs["client"] = client
+            return func(*args, **kwargs)
+
+        logger.warning(
+            "Creating a Nominal Client without using '--profile' is deprecated! "
+            "See 'https://docs.nominal.io/python/profile-migration' for details..."
+        )
```

Comment:
ehhh its not necessarily deprecated to pass base_url and token but i also don't really care cuz i never use this CLI